site/content/how-to/traffic-management/client-settings.md

```diff
@@ -0,0 +1,283 @@
+---
+title: "Client Settings Policy"
+weight: 800
+toc: true
+docs: "DOCS-000"
+---
+
+Learn how to use the `ClientSettingsPolicy` API.
+
+## Overview
+
+The `ClientSettingsPolicy` API allows Cluster Operators and Application Developers to configure the connection behavior between the client and NGINX.
+
+The settings in `ClientSettingsPolicy` correspond to the following NGINX directives:
+
+- [`client_max_body_size`](<https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size>)
+- [`client_body_timeout`](<https://nginx.org/en/docs/http/ngx_http_core_module.html#client_body_timeout>)
+- [`keepalive_requests`](<https://nginx.org/en/docs/http/ngx_http_core_module.html#keepalive_requests>)
+- [`keepalive_time`](<https://nginx.org/en/docs/http/ngx_http_core_module.html#keepalive_time>)
+- [`keepalive_timeout`](<https://nginx.org/en/docs/http/ngx_http_core_module.html#keepalive_timeout>)
+
+`ClientSettingsPolicy` is an [Inherited PolicyAttachment](https://gateway-api.sigs.k8s.io/reference/policy-attachment/) that can be applied to a Gateway, HTTPRoute, or GRPCRoute in the same namespace as the `ClientSettingsPolicy`.
+
+When applied to a Gateway, the settings specified in the `ClientSettingsPolicy` affect all HTTPRoutes and GRPCRoutes attached to the Gateway. This allows Cluster Operators to set defaults for all applications using the Gateway.
+
+When applied to an HTTPRoute or GRPCRoute, the settings in the `ClientSettingsPolicy` affect only the route they are applied to. This allows Application Developers to set values for their applications based on their application's behavior or requirements.
+Settings applied to an HTTPRoute or GRPCRoute take precedence over settings applied to a Gateway.
+
+TODO: link to general doc on policy inheritance
+
+This guide will show you how to use the `ClientSettingsPolicy` API to configure the client max body size for your applications.
+
+## Setup
+
+- [Install]({{< relref "/installation/" >}}) NGINX Gateway Fabric.
+- [Expose NGINX Gateway Fabric]({{< relref "installation/expose-nginx-gateway-fabric.md" >}}) and save the public IP address and port of NGINX Gateway Fabric into shell variables:
+
+   ```text
+  GW_IP=XXX.YYY.ZZZ.III
+  GW_PORT=<port number>
+  ```
+
+  {{< note >}}In a production environment, you should have a DNS record for the external IP address that is exposed, and it should refer to the hostname that the gateway will forward for.{{< /note >}}
+
+- Create the coffee and tea example applications:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.3.0/examples/client-settings-policy/app.yaml
```

Comment:
We should probably get a consistent pattern on this stuff. Some guides reference files, some guides use literal YAML.

Reply:
Ya, consistency would be good. I chose this approach for the setup files because they aren't the subject of the tutorial; they are just prerequisites. I used the YAML code blocks for the policies so the reader can see what the policy looks like.

Reply:
As a point of note, I'm looking into if it's possible YAML files could be embedded into pages using a shortcode.

Not useful for small examples, but great for single-sourcing bigger files.